service/collector/vmss.go

```diff
@@ -138,7 +145,22 @@ func (u *VMSS) Collect(ch chan<- prometheus.Metric) error {
 			}
 			var writes int64
 			{
-				writes, err = strconv.ParseInt(r.Response().Header.Get("x-ms-ratelimit-remaining-subscription-writes"), 10, 64)
+				future, err := vmssClient.CreateOrUpdate(ctx, key.ResourceGroupName(cr), *vmss.Name, vmss)
```

Comment:
I am not sure if this is a good idea. IIRC metrics collection can run in parallel in the operator to its normal execution.
There might be conflicts where you get the information and the operator does something with it before you createOrUpdate here. 

I think a mutex lock would be necessary for that.